staging: yocto-cfg-fragments: include devmapper configs in docker

hashraf91 opened the pull request:
Adds following patch for yocto-cfg-fragments:

* cfg/docker.cfg enable pre-req of CONFIG_DM_THIN_PROVISIONING

    CONFIG_MD and CONFIG_BLK_DEV_DM are pre-requisite of
    CONFIG_DM_THIN_PROVISIONING.

JIRA-ID: SB-24462

hashraf91 committed:
staging: yocto-cfg-fragments: include devmapper configs in docker

Adds following patch for yocto-cfg-fragments:

* cfg/docker.cfg enable pre-req of CONFIG_DM_THIN_PROVISIONING

    CONFIG_MD and CONFIG_BLK_DEV_DM are pre-requisite of
    CONFIG_DM_THIN_PROVISIONING.

JIRA-ID: SB-24462

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

ahsanhussain commented:
While you're at it can you also add CONFIG_EXT4_FS_SECURITY. This is not a requirement for runc/docker but some docker images expect this from the host FS driver, and fail to run containers without. This is enabled on other ewaol distros usually

ahsanhussain commented:
I had to rebuild flex kernel a bunch of times because of this

hashraf91 committed:
distro: virtualization-layer: enable ext4-fs support in kernel

Some docker images expect CONFIG_EXT4_FS_SECURITY from host FS driver

Suggeted-by: Ehson Hussain <ahsan.hussain@siemens.com>
Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 commented:
@ahsanhussain updated the PR, please check now.
